libs/orders/src/lib/components/order-data-provider/order-data-provider.ts

```diff
@@ -131,7 +131,8 @@ export const update = <T extends Omit<OrderFieldsFragment, 'market'> & Cursor>(
 ): T[] => {
   const updatedData = data ? [...data] : ([] as T[]);
   delta.forEach((orderUpdate) => {
-    const index = data?.findIndex((order) => order.id === orderUpdate.id) ?? -1;
+    const index =
+      data?.findIndex((order) => order?.id === orderUpdate.id) ?? -1;
```

Comment:
Why these changes the seem completely unrelated to the rest of this PR? Are the types incorrect here and order can be undefined?

Reply:
weirdly while I was testing i got some exceptions on stagnet1 and order was undefined

Reply:
<img width="1680" alt="image" src="https://github.com/vegaprotocol/frontend-monorepo/assets/16125548/d4bf9e71-0a10-4bf7-9d25-f81c59cc5823">

Reply:
added that "?." since it is an extra check that does not hurt and prevents runtime exceptions... but yeah, spotted that on staging only

Reply:
Its very annoying to have to clutter the code with syntax to account for poorly typed objects. Its fine for now. @bglownia we probably want to take a look at this